tests/test_seq2pat_batch.py

```diff
@@ -0,0 +1,284 @@
+                    [11, 12, 13, 1],
+                    [11, 13, 1],
+                    [12, 13, 1]]
+
+        counter = list_to_counter(patterns)
+
+        self.assertEqual(counter, collections.Counter({(11, 12): 1, (11, 12, 13): 1, (11, 13): 1, (12, 13): 1}))
+
+    def test_counter_to_list(self):
+        counter = collections.Counter({(11, 12): 1, (11, 12, 13): 2, (11, 13): 1, (12, 13): 1})
+
+        results = counter_to_list(counter, min_row_count=2)
+
+        self.assertEqual(results, [[11, 12, 13, 2]])
+
+    def test_batch_patterns_default_equal(self):
+        # Seq2Pat over 3 sequences
+        seq2pat = Seq2Pat(sequences=[["A", "C", "B", "A", "D"],
+                                     ["C", "B", "A"],
+                                     ["C", "A", "C", "D"]])
+
+        # Patterns
+        patterns = seq2pat.get_patterns(min_frequency=1.0)
+
+        seq2pat = Seq2Pat(sequences=[["A", "C", "B", "A", "D"],
+                                     ["C", "B", "A"],
+                                     ["C", "A", "C", "D"]], batch_size=2)
+
+        # Patterns
+        batch_patterns = seq2pat.get_patterns(min_frequency=1.0)
+
+        self.assertEqual(patterns, batch_patterns)
+
+    def test_batch_patterns_constraints_equal(self):
+        # Seq2Pat over 3 sequences
+        seq2pat = Seq2Pat(sequences=[["A", "C", "B", "A", "D"],
+                                     ["C", "B", "A"],
+                                     ["C", "A", "C", "D"]])
+
+        # Attribute - I: Price
+        price = Attribute(values=[[5, 5, 3, 8, 2],
+                                  [1, 3, 3],
+                                  [4, 5, 2, 1]])
+
+        # Attribute - II: Timestamp
+        timestamp = Attribute(values=[[1, 1, 2, 3, 3],
+                                      [3, 8, 9],
+                                      [2, 5, 5, 7]])
+
+        # Add Constraint
+        avg_constraint = 3 <= price.average() <= 4
+        seq2pat.add_constraint(avg_constraint)
+
+        # Patterns
+        patterns = seq2pat.get_patterns(min_frequency=0.8)
+
+        seq2pat = Seq2Pat(sequences=[["A", "C", "B", "A", "D"],
+                                     ["C", "B", "A"],
+                                     ["C", "A", "C", "D"]], batch_size=2)
+
+        # Attribute - I: Price
+        price = Attribute(values=[[5, 5, 3, 8, 2],
+                                  [1, 3, 3],
+                                  [4, 5, 2, 1]])
+
+        # Attribute - II: Timestamp
+        timestamp = Attribute(values=[[1, 1, 2, 3, 3],
+                                      [3, 8, 9],
+                                      [2, 5, 5, 7]])
+
+        # Add Constraint
+        avg_constraint = 3 <= price.average() <= 4
+        seq2pat.add_constraint(avg_constraint)
+
+        # Patterns
+        batch_patterns = seq2pat.get_patterns(min_frequency=0.8)
+
+        self.assertEqual(patterns, batch_patterns)
+
+    def test_input_no_upper_constraint_batch(self):
```

Comment:
Can you tell a bit on why we're trying to replicate that output?

Reply:
This is to check Seq2Pat can give the same results as we run the C++ package. So we know the Cython part is running correctly.